### PR TITLE
Adding OpenTelemetry collector/exporter for dev-env

### PR DESCRIPTION
**Pull request by lipoja**
- update of docker-compose.yml and podman-compose.yml
- added Jaeger to collect traces

[CLOUDDST-20308]

### CODE REVIEW

**Review by yashvardhannanavati — DISMISSED** [review]
The changes look good to me. I am not able to test it on my local because of some issue specific to my system. Could you please confirm if this was tested locally?

File: README.md [hunk 1]
```diff
@@ -283,10 +283,13 @@ environment variables.
   This defaults to `False`. If this is set to `True`, `OTEL_EXPORTER_OTLP_ENDPOINT` and `OTEL_SERVICE_NAME`
   must be set.
 * `OTEL_EXPORTER_OTLP_ENDPOINT` - The endpoint for the OpenTelemetry exporter.
-* `OTEL_SERVICE_NAME` - "iib-api"
+* `OTEL_SERVICE_NAME` - "iib-api" / "iib-worker"
 
 For more info on these environment variables, please refer to [Opentelemetry Guide](https://opentelemetry.io/docs/concepts/sdk-configuration/otlp-exporter-configuration/)
 
+Dev-env supports Opentelemtry tracing by default. It uses Jaeger for collecting and exporting traces.
```
Comment (JAVGan) [inline README.md:290]:
```suggestion
Dev-env supports Opentelemetry tracing by default. It uses Jaeger for collecting and exporting traces.
```

File: compose-files/docker-compose.yml [hunk 2]
```diff
@@ -118,7 +133,12 @@ services:
       IIB_DEV: 'true'
       REGISTRY_AUTH_FILE: '/root/.docker/config.json'
       REQUESTS_CA_BUNDLE: /etc/pki/tls/certs/ca-bundle.crt
-    # Make this privileged to be able to build container images
+      # Make this privileged to be able to build container images
+      # enable OpenTelemetry for dev-env
+      IIB_OTEL_TRACING: 'true'
+      OTEL_EXPORTER_OTLP_ENDPOINT: http://jaeger:4318
+      OTEL_EXPORTER_OTLP_TRACES_INSECURE: 'true'
+      OTEL_SERVICE_NAME: "iib-api"
```
Comment (zanssa) [inline compose-files/docker-compose.yml:141]: Should this be "iib-worker" ?

Comment (lipoja) [reply compose-files/docker-compose.yml]: Thanks, fixed

**Review by zanssa — APPROVED** [review]

**Review by JAVGan — APPROVED** [review]

**Pull request merged**